test/comms_serial_connection_tests/tempsvoltagestest.py

```diff
@@ -0,0 +1,392 @@
+    t_microcontroller_mV = 5068
+    t_LED_mV = 3000
+    t_PSU_mV = 20242
+    t_spindle_speed_monitor_mV = 1050
+
+    def give_me_a_PCB(outerSelf):
+
+        class YETIPCB(MockSerial):
+
+            @serial_query("?")
+            def do_something(self):
+                return outerSelf.status
+
+        return YETIPCB
+
+
+    def status_and_PCB_constructor(self, case=None, 
+                        pcb_temp=0,
+                        motor_driver_temp=0,
+                        transistor_heatsink_temp=0,
+                        microcontroller_mV=0,
+                        LED_mV=0,
+                        PSU_mV=0,
+                        spindle_speed_monitor_mV=0
+                        ):
+
+        # Use this to construct the test status passed out by mock serial object
+
+        if case == 1 or case==None:
+
+            self.status = "<Idle|MPos:0.000,0.000,0.000|Bf:35,255|FS:0,0|Pn:PxXyYZ|WCO:-166.126,-213.609,-21.822>"
+
+        elif case == 2:
+
+            self.status = "<Idle|MPos:0.000,0.000,0.000|Bf:35,255|FS:0,0|Ld:0|TC:" + \
+                str(pcb_temp) + "," + \
+                str(motor_driver_temp) + "|V:" + \
+                str(microcontroller_mV) + "," + \
+                str(LED_mV) + "," + \
+                str(PSU_mV) + "," + \
+                str(spindle_speed_monitor_mV) + ">"
+
+        elif case == 3:
+
+            self.status = "<Idle|MPos:0.000,0.000,0.000|Bf:35,255|FS:0,0|Ld:0|TC:" + \
+                str(pcb_temp) + "," + \
+                str(motor_driver_temp) + "," + \
+                str(transistor_heatsink_temp) + "|V:" + \
+                str(microcontroller_mV) + "," + \
+                str(LED_mV) + "," + \
+                str(PSU_mV) + "," + \
+                str(spindle_speed_monitor_mV) + ">"
+
+
+        # Need to construct mock PCB after the status, otherwise it'll run something else:
+        self.serial_module.s = DummySerial(self.give_me_a_PCB())
+        self.serial_module.s.fd = 1 # this is needed to force it to run
+        self.serial_module.start_services(1)
+        sleep(0.01)
+
+    def setUp(self):
+        # If the set_up method raises an exception while the test is running, 
+        # the framework will consider the test to have suffered an error, 
+        # and the runTest (or test_X_Name) method will not be executed.
+
+        self.m = Mock()
+        self.sm = Mock()
+        self.sett = Mock()
+        self.l = localization.Localization()
+        self.jd = Mock()
+
+        self.serial_module = serial_connection.SerialConnection(self.m, self.sm, self.sett, self.l, self.jd)
+        # self.serial_module.VERBOSE_ALL_RESPONSE = True
+
+
+    def tearDown(self):
+      self.serial_module.__del__()
+
+
+    def test_does_serial_think_its_connected(self):
+        """Test that serial module thinks it is connected"""
+        self.status_and_PCB_constructor()
+        assert self.serial_module.is_connected() == True, 'not connected'
```

Comment:
for each assertion, could be changed to

`assert self.serial_module.is_connected(), 'not connected'`

Reply:
oooh yisss, much tidier